xml_converter/src/xml_converter.cpp

```diff
@@ -62,48 +62,62 @@ vector<string> get_files_by_suffix(string directory, string suffix) {
     return files;
 }
 
-set<string> read_taco_directory(
+map<string, vector<string>> read_taco_directory(
     string input_path,
     map<string, Category>* marker_categories,
     vector<Parseable*>* parsed_pois) {
-    set<string> top_level_categories;
+    map<string, vector<string>> category_to_filepath;
```

Comment:
Can we add a more descriptive name here. This is only storing the top level paths and mapping to the locations that they are saved in. Maybe `top_level_categories_to_file_locations` or something like that. Maybe just `top_level_category_file_locations` even.